crates/stackable-operator/src/builder/pod/container.rs

```diff
@@ -21,11 +22,25 @@ pub enum Error {
         source: validation::Errors,
         container_name: String,
     },
+
+    #[snafu(display(
+        "The volumeMount is clashing with an already existing volumeMount with the same mountPath but a different content. \
+            The existing volumeMount is {existing_volume_mount:?}, the new one is {new_volume_mount:?}"
+    ))]
+    ClashingVolumeMountMountPath {
+        existing_volume_mount: Box<VolumeMount>,
+        new_volume_mount: Box<VolumeMount>,
+    },
 }
 
 /// A builder to build [`Container`] objects.
 ///
 /// This will automatically create the necessary volumes and mounts for each `ConfigMap` which is added.
+///
+/// This struct is often times using an [`IndexMap`] to have consistent ordering (so we don't produce reconcile loops).
+/// We are also choosing it over an [`std::collections::BTreeMap`], as it's easier to debug for users, as logically
+/// grouped volumeMounts (e.g. all volumeMounts related to S3) are near each other in the list instead of "just" being
+/// sorted alphabetically.
```

Comment:
```suggestion
/// This struct is often times using an [`IndexMap`] to have consistent ordering (so we don't produce reconcile loops).
/// We are also choosing it over a [`BTreeMap`], because it is easier to debug for users, as logically
/// grouped volumeMounts (e.g. all volumeMounts related to S3) are near each other in the list instead of "just" being
/// sorted alphabetically.
```

I removed the qualifier for `BTreeMap` to make it consistent with the comment on `pod::PodBuilder`.
If that doesn't work in the docs, the both `BTreeMap` and `IndexMap` should have the qualifiers (on both comments).

Reply:
Yeah, I tried to avoid importing it behind #[cfg(docs)], but let's keep the rustdocs clear. I will fix any errors popping up